### PR TITLE
Add taxcalc/tests/test_records.py file to increase tested code coverage

### PR DESCRIPTION
**Pull request by martinholmer**
This pull request adds some tests of the Records class in order to increase the test suite's code coverage.
There is no change in tax logic or data.

### CODE REVIEW

File: taxcalc/records_variables.json [hunk 1]
```diff
@@ -4,85 +4,85 @@
       "required": true,
       "type": "int",
       "desc": "Unique positive numeric identifier for filing unit",
-      "form": {"2017": "private info"}
+      "form": {"2017-2017": "private info"}
```
Comment (sebastiansajie) [inline taxcalc/records_variables.json]: @martinholmer 
Could you please explain why you have changed the format for "form" in records_variables.json from 2017 to 2017-2017.

**Comment by sebastiansajie** [comment]
@martinholmer
Could you please explain why you have changed the format for "form" in records_variables.json from 2017 to 2017-2017. The tax form shows it as 2017-18, so 2017-2018 could have been appropriate.

**Comment by martinholmer** [comment]
@sebastiansajie asked in PR#62:
> Could you please explain why you have changed the format for "form" in `records_variables.json` from 2017 to 2017-2017. The tax form shows it as 2017-18, so 2017-2018 could have been appropriate.

I thought we had settled on using the first year of an assessment year as the tax year in the India tax-calculator.
So, 2017 stands for AY2017-18 and 2018 stands for AY2018-19.

Right now, we have data only for 2017 (ie, AY2017-18).  When TPRU moves to using 2018 (ie, AY2018-19) data, then the `records_variables.json` "form" fields can be updated.  If a variable is on the same line in both years, it is economical to specify the key as "2017-2018" signifying that the variable appears on the same line in both AY2017-18 and AY2018-19.  If the variable is on a different line, then the "form" entry would have two key-value pairs:  "2017-2017": "line ...", "2018-2018": line different ...".

This is admittedly potentially confusing because India's tax years are not calendar years like in the USA.
If you think its too confusing, I'd be happy to drop out the one test (in `test_records.py`) that enforces this notation.  (We enforce it in USA because we automatically generate written documentation from the `records_variables.json` file and so the JSON needs to have a standard format.)

**Comment by sebastiansajie** [comment]
@martinholmer, I have reopened this pull request that I had inadvertently closed.

**Comment by martinholmer** [comment]
@sebastiansajie said:
> I have reopened this pull request [#62] that I had inadvertently closed.

Thanks!   I'll revise it to eliminate the testing of the `records_variables.json` "form" year format used in the USA tax-calculator.

**Comment by martinholmer** [comment]
Pull request #62 has been revised so that the  `records_variables.json` "form" field and the `test_records_variables_contents` test do not conflict with India assessment year notation.

**Comment by martinholmer** [comment]
I believe PR #62 is now ready to merge as is PR #63, which is now consistent with the new test added in #62.

@indira-tpru @keshavchoudhary87 @adc2-tpru @sebastiansajie

**Comment by martinholmer** [comment]
@keshavchoudhary87 asked in PR #62:
>  In the file `taxcalc/tests/conftest.py` the fraction of sample is mentioned as 0.10 so does it limit the sample to this upper bound. And if we take a smaller sample size will this needs to be changed? 

The `configtest.py` file defines two different input data sets that can be used in the tests.  

First, it defines a `pit_fullsample` which contains **all** the records in the `pit.csv` file.

Second, it defines a `pit_subsample` which contains a **10% sample** of the records in the `pit.csv` file.

We often use the second smaller input data in order to make the tests execute faster.  
But some of the tests make use of the `pit_fullsample` data.

Right now, the `pit.csv` file contains only 100 records, so using a 10% sample seemed to be about the smallest that made sense.  When the `pit.csv` file contains hundreds of thousands of records, you will probably be able to make the tests run faster using a sampling rate of less than 5%.

Does this response answer you excellent questions?  If not, please ask more questions.

**Comment by keshavchoudhary87** [comment]
@martinholmer : Thanks for your lucid explanations. we were thinking of working with around 0.5% of the population which would be in 250,000. so should we let the pit_subsample be 0.1 or 0.05

**Comment by martinholmer** [comment]
@keshavchoudhary87 said in PR #62:
> Thanks for your lucid explanations. we were thinking of working with around 0.5% of the population which would be in 250,000. 

OK.  Eventually you'll have a `pit.csv` file with 250,000 records.  That seems reasonable.

> so should we let the pit_subsample be 0.1 or 0.05

No.  Turns out that my explanations were not that "lucid" because I've left you with the wrong impression.
Remember that the pit_subsample is used **only** in the tests; never in any app.

For now, leave the test pit_subsample at 10%, which is a sampling fraction of 0.10, because the `pit.csv` file contains only 100 records (not 250,000 records).  Later, when you move to using the much larger `pit.csv` file, you can lower the sampling fraction for the test pit_subsample below 0.10.  So, when your have 250,000 records in the `pit.csv` file, **for testing purposes** you could probably lower the test sampling fraction to something like 0.03 (3% percent), which would imply a test subsample of about 7,500 records.

Does this make sense?

@indira-tpru @adc2-tpru @sebastiansajie

**Comment by keshavchoudhary87** [comment]
@martinholmer : Thanks for the detailed explanation. Yes got it.

**Comment by sebastiansajie** [comment]
@martinholmer , why is this pull request still open? do you prefer not to go ahead with this?

**Comment by martinholmer** [comment]
@sebastiansajie asked in PR #62:
> why is this pull request still open? do you prefer not to go ahead with this?

I don't know why it is still open.  @keshavchoudhary87 asked several questions about it, but I think those questions have been answered satisfactorily.  I was expecting him to merge this pull request.

If I preferred to "not go ahead with this", I would close the pull request (without merging).

I also wonder why PR #58 is still not merged.  As I said [here](https://github.com/TPRU-demo/pitaxcalc-demo/issues/65#issuecomment-428285845) over two weeks ago, I can't respond to your request in issue #65 until the pending PRs are merged.

**Pull request merged**